umap/templates/umap/js.html

```diff
@@ -58,4 +58,5 @@
         <script src="{{ STATIC_URL }}umap/js/umap.share.js" defer></script>
         <script src="{{ STATIC_URL }}umap/js/umap.js" defer></script>
         <script src="{{ STATIC_URL }}umap/js/umap.ui.js" defer></script>
+        <script src="{{ STATIC_URL }}umap/js/umap.components.js" defer></script>
```

Comment:
I'm trying to understand how do we guarantee that map are loaded when all the context is ready. Do I understand correctly that this `umap.component` is expected to be included after all the other ones, so the dom gets parsed then only?

Reply:
If I understand correctly [this article](https://lamplightdev.com/blog/2021/03/15/when-are-the-constructor-and-connectedcallback-methods-called-when-creating-a-custom-element/), the `connectedCallback()` from the `umap-fragment` custom element will be called only when the DOM is evaluated within the page.

Even if I delay by 5sec the registration of the custom element:

`setTimeout(() => register(UmapFragment, 'umap-fragment'), 5000)`

Once loaded, the maps will eventually appear. I can even put the whole `UmapFragment` class definition within the timeout and it will still work (which is a bit magical!).

Reply:
The [Custom element lifecycle callbacks](https://developer.mozilla.org/en-US/docs/Web/API/Web_Components/Using_custom_elements#custom_element_lifecycle_callbacks)

Reply:
> So a `<script defer src="yourelement.js">` does the job; **but** will run **after** all DOM is created,
your components are now created (very) late. So you now have to deal with (more) FOUCs. — https://stackoverflow.com/a/70952159

Maybe something to consider, even for #1463